src/renderer/Widget/GmailWidget/renderer.tsx

```diff
@@ -0,0 +1,157 @@
+import React, {useEffect, useState} from 'react';
+import {NUM_DATA_DISPLAY, ERROR_MESSAGE} from './constant';
+import {Button, Spin, Alert, Tag} from 'antd';
+import {LoadingOutlined} from '@ant-design/icons';
+import {loadGAPI, initClient, getMailFromID, formatListMail, handleAuthorize, checkSignin, getListMail, signIn, signOut} from './service';
+import MailDetail from './MailDetail';
+import './style.css';
+
+const Gmail = ({apiKey = '', clientID = '', data = []}: {
+  apiKey?: string;
+  clientID?: string;
+  data?: GmailData[];
+}) => {
+  const [dataSource, setDataSource] = useState(data);
+  const [existGAPI, setExistGAPI] = useState(false);
+  const [dataDisplay, setDataDisplay] = useState([] as GmailData[]);
+  const [currentPape, setCurrentPage] = useState(0);
+  const [nextPageToken, setNextPageToken] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [signin, setSignin] = useState(false);
+  const [error, setError] = useState({logOut: '', invalidKey: ''});
+
+  useEffect(() =>{
+    if (dataSource.length === 0) setDataDisplay([]);
+    const startIndex = NUM_DATA_DISPLAY * currentPape;
+    if (dataSource.length <= startIndex) return;
+
+    let endIndex = startIndex + NUM_DATA_DISPLAY;
+    endIndex = endIndex > dataSource.length ? dataSource.length : endIndex;
+
+    const copyData = [...dataSource];
+    const currentData = copyData.length > NUM_DATA_DISPLAY ? copyData.slice(startIndex, endIndex) : copyData;
+    setDataDisplay(currentData);
+  }, [currentPape, dataSource]);
+  useEffect(() =>{
+    const handleSigninCallBack = (isSignedIn: boolean) => {
+      setSignin(isSignedIn);
+      prepareData(isSignedIn);
+    };
+    const handleGAPIClient = async () => {
+      if (!existGAPI) {
+        await loadGAPI();
+        setExistGAPI(true);
+      }
+      if (window.kintone && existGAPI) {
+        gapi.load('client:auth2', () => {
+          setLoading(true);
+          initClient(apiKey, clientID).then(() => {
+            setSignin(checkSignin());
+            handleAuthorize(handleSigninCallBack);
+            handleSigninCallBack(checkSignin());
+          }).catch(() => {
+            setError({invalidKey: ERROR_MESSAGE.API_KEY, logOut: ''});
+            setLoading(false);
+          });
+        });
+      }
+    };
+    handleGAPIClient();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Remove this disable comment. Wrap `prepareData` in `useCallback` to avoid infinite loop.